BookStore.Domain/Models/Book.cs

```diff
@@ -1,8 +1,14 @@
-﻿namespace BookStore.Domain.Models
+﻿using MongoDB.Bson;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace BookStore.Domain.Models
 {
+    [BsonIgnoreExtraElements]
     public class Book
     {
-        public string? Id { get; set; }
+        [BsonId]
+        [BsonRepresentation(BsonType.ObjectId)]
+        public string Id { get; set; } = string.Empty;
         public string? Title { get; set; }
         public string? Author { get; set; }
         public int PublicationYear { get; set; }
```

Comment:
I don’t think we will ever have a model with year as int. Use date for the publication date.